### ISSUE TITLE
What is a query, and how are top-1 and top-5 scored?

### ISSUE THREAD

**Issue by franklinmdev** [franklinmdev/docmatch#119]
Part of #111

## Question

What one query is, how many there are, and what top-1 and top-5 count.

Decide how a variant is generated from an entry and how many variants each entry carries, how the query set is split so nothing is fitted on what is scored, how many queries the table is measured over, what share have no correct entry and where those come from (freight, surcharges, one-off services, or entries held out of the catalog), and exactly what the separability diagnostic beside the table reports per arm.

Settled on the map already: out-of-catalog queries are in the set, top-1 and top-5 are over answerable queries only, and no operating threshold is chosen in Phase 3.

**Comment by franklinmdev** [comment]
## Decided

### What one query is

A query is one description text and nothing else, normalized by `metrics.normalization.normalize_text`, the same form #117 made the canonical catalog description. Nothing else rides in it: #117 fixed the entry at a SKU plus a canonical description, so there is no code, unit or price on either side to compare, and the map already holds whether the line code joins the query text as fog rather than a decision.

Truth is by construction. An answerable query carries the SKU of the entry it was generated from; an out-of-catalog query carries none.

No query carries two entries. #116 put splits and merges at about 1.4 percent and outside what is worth imitating, so the merged-reading case it handed to this ticket does not arise: the generator never produces one, rather than the query set excluding it after the fact.

### The query set

Every entry is in the set, and the catalog is whole, all 1,920 entries searchable, for every query in every slice.

Each entry carries one exact query, its canonical description unchanged, and two noisy variants. Out-of-catalog queries are drawn one per description, since there is no entry to cover and repeating a description buys nothing.

| | entries or descriptions | exact | noisy | queries |
|---|---|---|---|---|
| answerable, scored | 1,536 | 1,536 | 3,072 | 4,608 |
| answerable, development | 384 | 384 | 768 | 1,152 |
| out of catalog, scored | 813 | 271 | 542 | 813 |
| out of catalog, development | 203 | 68 | 135 | 203 |
| **total** | | **2,259** | **4,517** | **6,776** |

Out of catalog is 1,016 of 6,776, **0.150**, inside #114's 10 to 20 percent band.

**The table is measured over the 4,608 scored answerable queries**, with the separability diagnostic over those plus the 813 scored out-of-catalog queries.

### The exact rate is a weight, not a duplicated query

#116 measured that a reading reproduces its label exactly about two times in three, so the headline has to read as that mixture. An entry has exactly one exact form, so an exact slice larger than 1,920 could only be built by running byte-identical queries repeatedly, which buys no information and costs a third of every arm's run.

The headline is therefore the two slices weighted:

```
top-k = w * top-k over exact queries + (1 - w) * top-k over noisy queries,   w = 2/3
```

`w` is pinned in code as the constant `2/3`, inside #116's defensible 0.65 to 0.70 exact band, and printed in the report beside the table. It is a constant, not a knob: it is fixed before any arm runs and #122's pre-registration discipline applies to it. Every arm's row uses the same `w`.

Both slices carry every entry, so all 352 of #117's entries with a neighbour at Levenshtein 0.9 or above are in both, and the per-entry analysis #117 pointed at, where the gap between top-1 and top-5 lives, is whole rather than halved.

The exact slice is not a formality. A query byte-identical to its entry can still be lost: 352 entries have a near neighbour, and a vector arm can rank a near-duplicate above an exact match. Anything an arm gets wrong there is a genuine finding about that arm.

### The noise model

Four kinds, #116's whole recommendation, at its shares renormalized over the 75.9 percent of differing mass they carry:

| kind | #116 share | renormalized | queries |
|---|---|---|---|
| the reading carries extra words | 26.6% | 35.0% | 1,583 |
| a character or two of letters, neither side containing the other | 18.6% | 24.5% | 1,107 |
| digits dropped from the description | 16.1% | 21.2% | 958 |
| punctuation | 14.6% | 19.2% | 869 |

**A variant picks one kind and keeps it**, per #116's concentration table: a backend that drops the part number drops it on every row of that table, so error is a habit rather than a per-row die roll.

**Kinds are fixed; strength is calibrated to #116's bands.** Each kind's strength, how many characters are substituted, how much text is appended, how many digit-bearing words are dropped, is set so the noisy queries' normalized Levenshtein against their source entry reproduces #116's measured shape:

| band | target | source |
|---|---|---|
| [0.9, 1.0) | 36.2% | 72 of 199 differing pairs, blind pairing |
| [0.8, 0.9) | 9.5% | 19 of 199 |
| [0.7, 0.8) | 12.6% | 25 of 199 |
| [0.5, 0.7) | 9.5% | 19 of 199 |
| below 0.5 | 32.2% | 64 of 199 |

A test in the eval suite asserts both the kind shares and the band histogram, each band within 5 percentage points of its target. That turns "the noise model imitates measured backend error" from a claim in the README into a checked property. The calibration is against a train measurement, so #77 permits it and nothing is fitted on what is scored.

If the two constraints prove jointly unreachable, the band tolerance widens and the amount it widened by is recorded here rather than absorbed silently.

### The split

The 1,920 entries split 80/20 under one pinned seed, 1,536 scored and 384 development. The out-of-catalog draw splits the same way, stratified so both slices keep the one-to-two exact and noisy proportion.

Every knob #120 sets, candidate depth per arm, rerank depth, `ef_search`, anything the fusion needs beyond #113's k = 60 from the original paper, is settled on the development slice. The scored number is produced once.

#77 was already satisfied without this: catalog, queries and noise model all come from train, and the fixed subset is untouched. This is the stricter guarantee, that the knobs are not tuned on the queries the README's table is measured over.

**What the split does not isolate.** The catalog is whole in both slices, so a development entry can be the near-duplicate neighbour of a scored entry. The isolation is of queries, not of neighbourhoods. Making it neighbourhoods would mean splitting the catalog, which would change what every arm searches and make the two slices different problems.

### Scoring

An answerable query is **correct at top-1** when the top-ranked entry's SKU equals the source entry's SKU, and **correct at top-5** when that SKU is among the first five. Out-of-catalog queries never enter the headline, which is the map's answerable-only rule.

**Ties are broken by SKU ascending, and the rank-1 tie rate is printed per arm.** Ties are designed into this benchmark rather than incidental: digit-drop noise is 21.2 percent of variants, and 115 of #117's 287 near-duplicate pairs at 0.9 differ only in their digits, so a query whose digits were dropped can be exactly equidistant from both members of a pair. The trigram and fused arms will produce these; the vector and rerank arms score in floats and almost never will.

The SKU is 8 hex of the SHA-256 of the description (#117), so ordering by it is pinned and pseudorandom, independent of which entry is correct. A two-way tie scores correct half the time in expectation, with no thumb on the scale. The cost is that a single run's headline carries the luck of the hash, worth a few tenths of a point at these counts, which is why the tie rate is reported rather than buried.

The two alternatives were rejected for making the scorer decide the comparison: counting a tie as a miss penalizes the trigram arm for integer scoring rather than for retrieval quality, and counting the tied block as correct would give a perfect top-1 to an arm that ties the whole catalog.

### The separability diagnostic

Per arm, over the 4,608 scored answerable and 813 scored out-of-catalog queries, using each query's top-1 score:

| column | what it is |
|---|---|
| reject at 0.99 | share of out-of-catalog below the score cut that keeps 0.99 of answerable |
| reject at 0.95 | the same at 0.95 |
| reject at 0.90 | the same at 0.90 |
| AUROC | answerable positive, out of catalog negative |

The four arms' scores are not commensurable, trigram similarity and cosine on 0 to 1, a fused reciprocal-rank sum around 0.03, cross-encoder logits, so nothing may compare raw scores across rows. Anchoring each cut to a quantile of that arm's **own** answerable distribution makes the rows comparable although the scores are not, and AUROC is rank based and therefore scale free by construction.

No operating threshold is chosen. The whole curve is printed and the cut is a reporting device; picking a point on it is Phase 4 routing's job, as the map settled. #122 gets the number it needs to weigh whether the reranker rejects out-of-catalog queries better.

## Derived while resolving, not put to the tracker

Each of these follows from a decision above or from a closed ticket, and none had a live alternative worth a round.

1. **The query is normalized before retrieval**, the same `normalize_text` form as the canonical description, because a real pipeline normalizes a reading before it searches. This is why spacing noise is unobservable: it normalizes back to the entry and the query would become an exact one. The fourth kind is therefore **punctuation**, which survives normalization, and never spacing alone. #116 had already set case and whitespace aside for the same reason.
2. **The bleed source for "extra words" is never a catalog entry.** #116 found half of the longer readings carry another labeled cell of the same row, a column bleeding in, and an entry here has no row to bleed from. The appended text is a numeric or unit-shaped token for that half, standing for a quantity, price or unit cell, and a fragment of a train singleton description for the other half, standing for page text. Appending another catalog entry's text would manufacture the merged reading carrying two entries that #116 ruled out, so the pool the bleed is drawn from is non-catalog by construction. Median growth follows #116 at roughly 14 to 32 characters.
3. **An entry's two variants draw their kinds independently.** #116 ties the habit to a document and #114 ties a variant to a vendor, so two variants stand for two documents and a collision on the same kind is what independence should produce, not something to design out.
4. **The out-of-catalog population is weighted the same way as the answerable one** wherever a rate is computed over it, `w = 2/3` on its exact slice, so the two populations in the separability diagnostic are weighted alike.
5. **One pinned seed** covers the entry split, the out-of-catalog draw, the kind draw and the strength draw. Which flag exposes it is #121's call.
6. **The out-of-catalog guard is 0.95, deliberately loose.** A guard exists only to keep a singleton that is the same item respelled out of the unanswerable population. It should not remove near misses: #117 showed a neighbour at 0.9 is usually a genuinely different item, 115 of its 287 pairs differing only in digits, and those near misses are what the separability diagnostic exists to measure. The draw is uniform over distinct descriptions rather than weighted by occurrence, for the same reason the answerable side is not traffic weighted.
7. **No glossary change.** #117 settled "catalog entry" and "SKU" without touching `CONTEXT.md`, and the map carries no Phase 3 terms, so the vocabulary lands with the spec rather than ticket by ticket as Phase 2's map did.

## Measured for this ticket

Probes in `notes/probes/query-set/`, untracked, over DocILE train. Counts and shares only; no description text left the machine.

**Occurrences.** The probe reproduces #117 exactly, 1,920 entries out of 8,855 distinct normalized descriptions over 5,180 documents, which is the check that this ticket and #117 are counting the same thing.

| | value |
|---|---|
| line occurrences of catalog entries | 15,254 |
| line occurrences of singletons | 11,527 |
| occurrences per entry, p50 / mean / p95 / max | 3 / 7.94 / 20 / 2,060 |
| documents per entry, p50 / p95 / max | 3 / 11 / 125 |
| share of occurrences carried by the top entry | 0.135 |
| by the top 10 / 100 / 500 entries | 0.268 / 0.505 / 0.745 |

This is what rules out a traffic-weighted query set. One description would be 13.5 percent of the table and 100 of 1,920 would decide half of it, so the headline would measure a handful of descriptions rather than a catalog.

**How close a singleton sits to the catalog.** Nearest catalog entry by normalized Levenshtein, over all 6,935 singletons:

| band | singletons |
|---|---|
| [0.95, 1.00) | 221 |
| [0.90, 0.95) | 272 |
| [0.85, 0.90) | 175 |
| [0.80, 0.85) | 215 |
| [0.70, 0.80) | 445 |
| below 0.70 | 5,607 |

Guards leave 6,714 survivors at 0.95, 6,442 at 0.90, 6,267 at 0.85 and 6,052 at 0.80. The pool is abundant at every guard, so the guard costs nothing and was set by what keeps the truth correct rather than by what the draw needs.

## Caveats

- #116's bias carries into the noise model. 11 to 16 percent of labeled descriptions sit on rows that pair on nothing, and a badly read row is the row least likely to pair, so the 0.65 to 0.70 exact band is an upper bound and the real low tail is thicker. `w = 2/3` is therefore, if anything, generous to every arm equally.
- The band targets come from 199 differing pairs. #116 called any bucket under 10 rows indicative only; the two smallest bands here hold 19 each, which is why the tolerance is 5 points rather than tighter.
- 1,016 out-of-catalog queries against #114's band, which #114 itself called a design choice rather than a measurement, since no source quantifies freight and one-off service lines per invoice.
- The out-of-catalog queries are real invoice line text, so they carry whatever DocILE labels carry. They are drawn from train, never val, so the fixed subset stays out, and nothing in the query set touches it.

## What this leaves for other tickets

- **#120** owns candidate depth, the fusion, rerank depth and the latency column, and now knows the set it runs over is 6,776 queries, 5,421 of them scored.
- **#121** owns the command, the seed flag, and which of these numbers print beside the table: the headline, the exact and noisy slices, the four kinds, the band histogram, the rank-1 tie rate per arm and the separability columns.
- **#122** gets the separability columns to weigh against latency, and the note that `w`, the band tolerance and the tie rule are all pinned before any arm runs, so its own pre-registration is not the only thing standing between the table and a fitted decision.

**Comment by franklinmdev** [comment]
Resolved above. The query set, its noise model, its split, the scoring rule and the separability diagnostic are settled; #120 and #121 now have the set they run over.
